google/cloud/spanner/value_test.cc

```diff
@@ -937,8 +938,17 @@ TEST(Value, CommitTimestamp) {
   EXPECT_FALSE(bad.ok());
 }
 
+// This TEST_P fixture expects a "Param" that is a std::tuple containing the
+// following members:
+//
+// 1. The test case's name. This is for gtest and needs to be unique.
+// 2. The `Value` instance to stringify.
+// 3. The expect stringified string value
+// 4. An optional lambda (nullptr is allowed) that may apply IO manipulators to
+//    the ostream that may affect the rendered output of the stringified `Value.
 class ValueStreamOperator
-    : public TestWithParam<std::tuple<std::string, Value, std::string>> {
+    : public TestWithParam<std::tuple<std::string, Value, std::string,
+                                      std::function<void(std::ostream&)>>> {
```

Comment:
Might it be nicer to make this non-nullable and change to `std::function<std::ostream(std::ostream&)>`?

Reply:
Done.